File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        int res;
        for (string c : tokens) {
            if(c == "+") {
                int b = s.top();
                s.pop();
                int a = s.top();
                s.pop();
                s.push(a + b);
            } else if (c == "-") {
                int b = s.top();
                s.pop();
                int a = s.top();
                s.pop();
                s.push(a - b);
            } else if (c == "*") {
                int b = s.top();
                s.pop();
                int a = s.top();
                s.pop();
                s.push(a * b);
            } else if (c == "/") {
                int b = s.top();
                s.pop();
                int a = s.top();
                s.pop();
                s.push(a / b);
            } else {
                 s.push(stoi(c));
            }
               
                    
             
        }

        return s.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if token == "+":
                val2 = stack.pop()
                val1 = stack.pop()
                stack.append(val1 + val2)
            elif token == "-":
                val2 = stack.pop()
                val1 = stack.pop()
                stack.append(val1 - val2)
            elif token == "*":
                val2 = stack.pop()
                val1 = stack.pop()
                stack.append(val1 * val2)
            elif token == "/":
                val2 = stack.pop()
                val1 = stack.pop()
                res = val1 / val2

                if res >= 0:
                    res = math.floor(res)
                else:
                    res = math.ceil(res)
                stack.append(res)

            else: 
                stack.append(int(token))

        return stack.pop()